docs/architecture/adr-002-event-manager-contract.md

```diff
@@ -0,0 +1,94 @@
+# ADR 002: POAP Manager Contract
+
+## Changelog
+
+- July 11, 2022: Initial draft started;
+- July 12, 2022: Initial draft finished;
+- July 14, 2022: First review;
+- July 15, 2022: Second review.
+
+## Status
+DRAFTED
+
+## Abstract
+This ADR defines the architecture of the POAP Manager contract. This contract give a user the possibility to manage
+the minting process of a POAP by setting a range of conditions that users needs to fulfill in order to be able to claim
+the POAP.
+
+## Context
+Inside Desmos based social networks, users have a profile with which they interact with the different features that the
+applications provides. From the `v4.1.0` release, users can also create contents on-chain through the post module.
+Looking forward to the next Cosmoverse 2022, we would like to have a way to be able to check the eligibility to mint a POAP. Attendees that wants to receive the badge, will be required to fulfill two simple tasks:
+1. Comment the Cosmoverse post we previously created
+2. React to the Cosmoverse post
+
+Once done, the contract will trigger the mint of the POAP automatically. The mint will take place separately inside the
+`POAP-Contract` and no one else besides the `POAP-Manager` will be able to perform the operation.
+
+## Decision
+The idea is to write a simple contract that make usage of the recently tagged `v1.0.0` of the `Desmos-Bindings` package
+to check on the 2 mint conditions and then call the `POAP-contract` to mint the POAP.
+
+## Specifications
+Here below the specifications for the contract's messages:
+
+### Messages
+
+#### Instantiate
+```rust
+pub struct InstantiateMsg {
+  pub admin: Addr,
```

Comment:
```suggestion
  pub admin: String,
```

Reply:
@bragaz  InstantiateMsg admin address is not verified. So it should be `String`.